plugins/inputs/elasticsearch/elasticsearch.go

```diff
@@ -257,7 +306,65 @@ func (e *Elasticsearch) gatherClusterStats(url string, acc telegraf.Accumulator)
 	return nil
 }
 
-func (e *Elasticsearch) gatherData(url string, v interface{}) error {
+func (e *Elasticsearch) gatherClusterStats(url string, acc telegraf.Accumulator) error {
+	clusterStats := &clusterStats{}
+	if err := e.gatherJsonData(url, clusterStats); err != nil {
+		return err
+	}
+	now := time.Now()
+	tags := map[string]string{
+		"node_name":    clusterStats.NodeName,
+		"cluster_name": clusterStats.ClusterName,
+		"status":       clusterStats.Status,
+	}
+
+	stats := map[string]interface{}{
+		"nodes":   clusterStats.Nodes,
+		"indices": clusterStats.Indices,
+	}
+
+	for p, s := range stats {
+		f := jsonparser.JSONFlattener{}
+		// parse json, including bools and strings
+		err := f.FullFlattenJSON("", s, true, true)
+		if err != nil {
+			return err
+		}
+		acc.AddFields("elasticsearch_clusterstats_"+p, f.Fields, tags, now)
+	}
+
+	return nil
+}
+
+func (e *Elasticsearch) setCatMaster(url string) error {
```

Comment:
This function is confusing to me. Why doesn't it do the _cat info parsing itself rather than just appending a string to the struct and then making decisions somewhere else?